### ISSUE TITLE
Easy way to run models on a list of examples?

### ISSUE THREAD

**Issue by orionw** [castorini/rank_llm#117]
Hi there! Great work on these -- would love to try them out!

I was curious if there is already a method/process for reranking given a list of pairs -- it seems like I would need to build out a `Request` object with `candidates`, but I'm not 100% certain.  

Is there any minimal example scripts? Or perhaps one that loads from a JSON file that I can emulate? 

Currently the README only has the one example that is fairly non-intuitive for new data that isn't an existing TREC dataset: I have to dig through several layers to find out what the inputs and outputs of these functions are and it still isn't quite clear how I could load my own data into it.  

Any help would be greatly appreciated and I'm sure would be useful to the broader community as well!

**Comment by TaKneeAa** [comment]
Following!

**Comment by sahel-sh** [comment]
You are right, you need to create the request object with query and candidates.
But alternatively you can create it from a json (see `demo/rerank_inline_hits.py`. You can also load retrieved results from a json file (see `demo/rerank_stored_retrieved_results.py`)

**Comment by sahel-sh** [comment]
More examples are added under the demo folder
